core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel_Common.hpp

```diff
@@ -32,7 +32,7 @@ struct ParallelReduceCopy {
   // Copy the result back to device if the view is on the device.
   static void memcpy_result(PointerType dest, PointerType src, size_t size,
                             bool ptr_on_device) {
-    if (ptr_on_device) {
+    if (ptr_on_device && 0 < size) {
```

Comment:
this just look wrong.  so what if the size is zero you enter the else-branch and dereference the pointers?

Reply:
Fixed.